pkg/middleware/auth_proxy.go

```diff
@@ -15,11 +15,11 @@ var header = setting.AuthProxyHeaderName
 func logUserIn(auth *authproxy.AuthProxy, username string, logger log.Logger, ignoreCache bool) (int64, *authproxy.Error) {
 	logger.Debug("Trying to log user in", "username", username, "ignoreCache", ignoreCache)
 	// Try to log in user via various providers
-	id, err := auth.Login(logger, ignoreCache)
-	if err != nil {
-		logger.Error("Failed to login", "username", username, "message", err.Error(), "error", err.DetailsError,
+	id, e := auth.Login(logger, ignoreCache)
```

Comment:
In case other reviewers wonder about this:
I looked this up, `e` is an `*auth.Error`, not `error`.

Reply:
Yes, I suspect we should convert this code later to return an `error` implementation :)

Reply:
Yeah, I think that's out of scope though :)

Reply:
@sakjur Absolutely :D